### ISSUE TITLE
Cordova - Satellizer Google OAuth2.0 403 disallowed_useragent

### ISSUE THREAD

**Issue by slinkydeveloper** [sahat/satellizer#1094]
Hi,
I'm using Cordova/Ionic and Satellizer to develop an application that log with Google OAuth2.0. I'm having some problems with authentication on Android: After i start the authentication, I receive an error 403 disallowed_useragent. What can i do?

Thank you

**Comment by qdequippe** [comment]
+1 I have the same problem, I tried this https://github.com/RocketChat/Rocket.Chat.Cordova/issues/159#issuecomment-275730484 but now I have an error with the redirect_uri that I did not have before.

**Comment by SamuelRoberto** [comment]
I fixed this problem using this workaround.
Open your cordova's configuration file called "config.xml" and add the following line:

`<preference name="OverrideUserAgent" value="Mozilla/5.0 Google" />`

**Comment by qdequippe** [comment]
@SamuelRoberto and with this workaround you didn't have another error with the redirect_uri ?

**Comment by SamuelRoberto** [comment]
@qdequippe No, I didn't because it just change the UserAgent. Try that and tell me if it works.

**Comment by qdequippe** [comment]
@SamuelRoberto I already tried that and it failed due to missing `redirect_uri`. It's weird because for example Facebook login works without this `<preference name="OverrideUserAgent" value="Mozilla/5.0 Google" />` but it doesn't work if I add it to my "config.xml" (I have an error on `redirect_uri`, before to add it the `redirect_uri` is set to `localhost` and when I add it the `redirect_uri` is set to `file://`).

I set my `redirect_uri` here (I am using Ionic with Cordova)
```
if (ionic.Platform.is('ios') || ionic.Platform.is('android')) {
   $authProvider.cordova = true;
   commonConfig.redirectUri = 'http://localhost/';
}
```
I don't understand why

**Comment by SamuelRoberto** [comment]
@qdequippe Maybe you should try to remove the if statement because changing the UserAgent, Ionic doesn't recognize the right platform. Try that for test.

```
$authProvider.cordova = true;
commonConfig.redirectUri = 'http://localhost/';
```

If you can't do that in production you could add some variable in the Application project to recognize the platform.

**Comment by qdequippe** [comment]
@SamuelRoberto I didn't think about this solution and it works thanks!

**Comment by eladcandroid** [comment]
@SamuelRoberto I tried this solution but inow I don't see my google accounts and need to login by writing it manually.. did you handle with this?

**Comment by SamuelRoberto** [comment]
@eladcandroid Mmh I didn't try this, but I think that this could happen because you are "masking" your browser's User Agent.

Anyway I don't suggest you to use "satellizer" anymore if you are writing a new application that requires oAuth2 and Ionic-Cordova. There are a lot of better alternatives that you could approch use to do that.
